compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -10533,16 +10531,19 @@ OMR::Z::TreeEvaluator::arraycmpHelper(TR::Node *node,
       TR::Register *baseReg1Temp = NULL;
       TR::Register *baseReg2Temp = NULL;
 
+
+      // if any of the to be generated CLC instructions will need a large disp or the reference offsets are uneven, generate an LA up front.
+      if ((loopIters * TR_MAX_CLC_SIZE + baseSource1Ref->getOffset() > TR_MAX_SS_DISP)
```

Comment:
Just comparing this change to the original code, previously we would generate LA for the memory reference which would need the large displacement. Seems like now you will generate for both, even if second mem ref does not need. Can you explain why ?

Reply:
the challenge here was that if we have different displacements, comparison would be hard since there is no compare address instruction. I considered a few solutions:
1- Do nothing in that case.
2- `aTempRegister=Ref1Base-ref2Base` then `compare aTempRegister with (ref2offset - ref1offset)` and jump if equal.
3- use LA if there are non equal displacements.

I selected option 3 since we need it for large displacement anyways. Option 1 may not be a bad one since having two equal address nodes with different small offsets is an edge case perhaps.

Reply:
I see, can you add comment to the code explaining this ?